include/swift/AST/DiagnosticsSema.def

```diff
@@ -5310,8 +5310,8 @@ ERROR(vararg_not_allowed,none,
       "variadic parameter cannot appear outside of a function parameter list",
       ())
 ERROR(expansion_not_allowed,none,
-      "pack expansion %0 cannot appear outside of a function parameter list, "
-      "function result, tuple element or generic argument list", (Type))
+      "pack expansion repeat each %0 can only appear in a function parameter list, "
```

Comment:
Instead of writing 'repeat each' in the diagnostic, which looks funny: `repeat each 'T'`, can you create a PackExpansionType and pass it as %0 to the diagnostic? Then it will say `'repeat each T'`